packages/signer/src/authClient.ts

```diff
@@ -91,10 +276,48 @@ export class AuthClient {
     return this.completeAuthWithBundle({ bundle: credentialBundle, orgId });
   }
 
+  /**
+   * Initiates OAuth authentication with the specified provider.
+   * This method handles the complete OAuth flow including provider URL generation and response processing.
+   *
+   * @param {LoginWithOauthParams} params - Parameters for OAuth authentication
+   * @param {string} params.authProviderId - OAuth provider identifier (e.g., "google", "facebook")
+   * @param {string} [params.scope] - OAuth scope parameters
+   * @param {string} [params.claims] - OAuth claims parameters
+   * @param {Record<string, string>} [params.otherParameters] - Additional OAuth parameters
+   * @param {"popup" | "redirect"} [params.mode] - OAuth flow mode, defaults to "redirect"
+   * @returns {Promise<Signer>} Promise that resolves to an authenticated Signer instance
+   * @throws {Error} If OAuth flow fails or is cancelled
+   *
+   * @example
+   * ```ts
+   * const signer = await authClient.loginWithOauth({
+   *   type: "oauth",
+   *   authProviderId: "google",
+   *   mode: "popup"
+   * });
+   * ```
+   */
   public async loginWithOauth(params: LoginWithOauthParams): Promise<Signer> {
-    const authUrl = await notImplemented(params);
-    const response = await this.handleOauthFlow(authUrl);
+    const { targetPublicKey } = await this.getTekStamper();
+    const oauthConfig = await this.dev_request("prepare-oauth", {
```

Comment:
This may not work: we want to call `/prepare-oauth` on startup so we may need a separate method for that.

The reason for this is that browsers restrict when popup windows can be opened to stop spammy websites from opening a bunch of popups on load. Browser policies differ, but some examples are only allowing popups to open for a short time after a user operation or from contexts that the browser considers "downstream" from user operations. [Here's a thread](https://alchemyinsights.slack.com/archives/C082ZDA1822/p1755797329715369?thread_ts=1755796746.708609&cid=C082ZDA1822) talking about this.

If this is working in your tests, we can merge with this for now and revisit, but we'll want to check on this before releasing v5.